src/lib/optimizer/strategy/data_induced_predicate_rule.cpp

```diff
@@ -0,0 +1,130 @@
+
+        /**
+         * We have to check whether the expression is actually evaluable on the reducer_node. If this is not the case
+         * we have to swap the sides. This might be the case, because it is not guaranteed, that the predicates are
+         * ordered according to the input sides.
+         */
+
+        if (!expression_evaluable_on_lqp(reducer_side_expression, *reducer_node)) {
+          std::swap(reduced_side_expression, reducer_side_expression);
+        }
+        DebugAssert(!expression_evaluable_on_lqp(reducer_side_expression, *join_node->input(selection_side)),
+                    "Expected filtered expression to be uniquely evaluable on one side of the join");
+
+        const auto subquery = AggregateNode::make(
+            expression_vector(), expression_vector(min_(reducer_side_expression), max_(reducer_side_expression)),
+            reducer_node);
+
+        const auto min = ProjectionNode::make(expression_vector(min_(reducer_side_expression)), subquery);
+        const auto max = ProjectionNode::make(expression_vector(max_(reducer_side_expression)), subquery);
+
+        // The between predicate is built using the projected min and max from the reducer side expression.
+        auto between_predicate =
+            PredicateNode::make(between_inclusive_(reduced_side_expression, lqp_subquery_(min), lqp_subquery_(max)));
+        lqp_insert_node(join_node, selection_side, between_predicate);
+
+        const auto reduced_cardinality = estimator->estimate_cardinality(between_predicate);
+        lqp_remove_node(between_predicate);
+
+        // If the `MINIMUM_SELECTIVITY` is not met we will not consider this join_predicate for this side.
+        if (original_cardinality == 0 || (reduced_cardinality / original_cardinality) > MINIMUM_SELECTIVITY) {
+          return;
+        }
+        data_induced_predicates.emplace_back(join_node, selection_side, between_predicate);
+        /**
+         * Additionally to this, future work could focus on evaluating if a heuristic that inserts semi join reductions
+         * if no predicate is found could be of further interest. This could be done by searching for a predicate in the
+         * lqp and adding a semi join reduction if the minimum selectivity of the SJR rule is met. First attempts for
+         * this are in the commit history (commit f6c5fb5) of this PR.
+         *
+         * Searching for a predicate prior to the join must not ensure a good DIP performance. In contrast to that the
+         * source paper mentioned in this PR suggests Range Sets as a solution. This could improve the performance
+         * drastically.
+         */
+      };
+
+      // Having defined the lambda responsible for conditionally adding a data induced predicate, we now apply it to
+      // both inputs of the join. For outer joins, we must not filter the side on which tuples survive even without a
+      // join partner.
+      if (join_node->join_mode != JoinMode::FullOuter) {
```

Comment:
For the first question:
Is there a `RightOuter` join in hyrise? Because I couldn't find one in the `types.hpp`.

For the second question:
If you look at the definition of an anti join in hyrise, there is only the need to filter them for the left side, because there they rely on the existence of non-matching values on the right side.